apps/src/lib/node/ledger/mod.rs

```diff
@@ -739,6 +739,77 @@ async fn maybe_start_ethereum_oracle(
     }
 }
 
+/// This function runs `Shell::init_chain` on the provided genesis files.
+/// This is to check that all the transactions included therein run
+/// successfully on chain initialization.
+pub fn test_genesis_files(
+    config: config::Ledger,
+    genesis: config::genesis::chain::Finalized,
+    wasm_dir: PathBuf,
+) {
+    use namada::ledger::storage::mockdb::MockDB;
+    use namada::ledger::storage::Sha256Hasher;
+
+    use crate::facade::tendermint::block::Size;
+    use crate::facade::tendermint::consensus::params::ValidatorParams;
+    use crate::facade::tendermint::consensus::Params;
+    use crate::facade::tendermint::evidence::{Duration, Params as Evidence};
+    use crate::facade::tendermint::time::Time;
+    use crate::facade::tendermint::v0_37::abci::request;
+
+    // Channels for validators to send protocol txs to be broadcast to the
+    // broadcaster service
+    let (broadcast_sender, _broadcaster_receiver) = mpsc::unbounded_channel();
+
+    // Start dummy broadcaster
+    let _broadcaster = spawn_dummy_task(());
+
+    // craft a request to initailize the chain
+    let init = request::InitChain {
+        time: Time::now(),
+        chain_id: config.chain_id.to_string(),
+        consensus_params: Params {
+            block: Size {
+                max_bytes: 0,
+                max_gas: 0,
+                time_iota_ms: 0,
+            },
+            evidence: Evidence {
+                max_age_num_blocks: 0,
+                max_age_duration: Duration(Default::default()),
+                max_bytes: 0,
+            },
+            validator: ValidatorParams {
+                pub_key_types: vec![],
+            },
+            version: None,
+            abci: Default::default(),
+        },
+        validators: vec![],
+        app_state_bytes: Default::default(),
+        initial_height: 0u32.into(),
+    };
```

Comment:
this init chain request could be moved inside  of `shell::InitChainValidation::new`, IMO, since the caller isn't really interested in building a request comprised mostly of values that will be ignored anyway. the only parameter required seems to be the chain id, which can replace the init chain request fn arg. the genesis time can be acquired from inside the fn.